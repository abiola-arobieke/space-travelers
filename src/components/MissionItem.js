import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { joinMission, leaveMission } from '../redux/missions/missionSlice';
import 'bootstrap/dist/css/bootstrap.min.css';
import '../styles/Missions.css';

const MissionItem = ({ mission }) => {
  const dispatch = useDispatch();

  const handleJoinMission = () => {
    dispatch(joinMission(mission.mission_id));
  };

  const handleLeaveMission = () => {
    dispatch(leaveMission(mission.mission_id));
  };

  return (
    <tr className="row-cell">
      <td className="mission-td mission-name">{mission.mission_name}</td>
      <td className="mission-td mission-description">{mission.description}</td>
      <td className="mission-td mission-member">
        {mission.reserved ? (
          <button type="button" className="btn btn-primary active-button">Active Member</button>
        ) : (
          <button type="button" className="btn btn-secondary member-button">NOT A MEMBER</button>
        )}
      </td>
      <td className="mission-td mission-join">
        {mission.reserved ? (
          <button
            type="button"
            className="btn btn-outline-danger leave-button"
            onClick={handleLeaveMission}
          >
            Leave Mission
          </button>
        ) : (
          <button
            type="button"
            className="btn btn-outline-secondary join-button"
            onClick={handleJoinMission}
          >
            Join Mission
          </button>
        )}
      </td>
    </tr>
  );
};

MissionItem.propTypes = {
  mission: PropTypes.shape({
    mission_id: PropTypes.string.isRequired,
    mission_name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    reserved: PropTypes.bool.isRequired,
  }).isRequired,
};

export default MissionItem;
