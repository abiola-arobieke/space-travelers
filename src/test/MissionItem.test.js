import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import MissionItem from '../components/MissionItem';

// Mock Redux store
const mockStore = configureStore([]);
const store = mockStore({});

describe('MissionItem', () => {
  it('renders correctly', () => {
    const mission = {
      mission_id: '1',
      mission_name: 'Mission 1',
      description: 'Mission description',
      reserved: true,
    };

    const component = renderer.create(
      <Provider store={store}>
        <MissionItem mission={mission} />
      </Provider>,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
