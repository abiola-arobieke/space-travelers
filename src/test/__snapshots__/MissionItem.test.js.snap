// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionItem renders correctly 1`] = `
<tr
  className="row-cell"
>
  <td
    className="mission-td mission-name"
  >
    Mission 1
  </td>
  <td
    className="mission-td mission-description"
  >
    Mission description
  </td>
  <td
    className="mission-td mission-member"
  >
    <button
      className="btn btn-primary active-button"
      type="button"
    >
      Active Member
    </button>
  </td>
  <td
    className="mission-td mission-join"
  >
    <button
      className="btn btn-outline-danger leave-button"
      onClick={[Function]}
      type="button"
    >
      Leave Mission
    </button>
  </td>
</tr>
`;
